guides/v2.3/inventory/source-selection-algorithms.md

```diff
@@ -0,0 +1,167 @@
+* [ItemRequestInterface](https://github.com/magento-engcom/msi/blob/2.3.0-release/app/code/Magento/InventorySourceSelectionApi/Api/Data/ItemRequestInterface.php) represents the requested quantity for a specific SKU 
+* [SourceSelectionServiceInterface](https://github.com/magento-engcom/msi/blob/2.3.0-release/app/code/Magento/InventorySourceSelectionApi/Api/SourceSelectionServiceInterface.php) returns the source selection algorithm result for the specified `inventoryRequest`
+* [GetSourceSelectionAlgorithmListInterface](https://github.com/magento-engcom/msi/blob/2.3.0-release/app/code/Magento/InventorySourceSelectionApi/Api/GetSourceSelectionAlgorithmListInterface.php) returns the list of data interfaces that represent registered SSAs
+* [SourceSelectionAlgorithmInterface](https://github.com/magento-engcom/msi/blob/2.3.0-release/app/code/Magento/InventorySourceSelectionApi/Api/Data/SourceSelectionAlgorithmInterface.php) represents a single SSA
+* [SourceSelectionInterface](https://github.com/magento-engcom/msi/blob/2.3.0-release/app/code/Magento/InventorySourceSelectionApi/Model/SourceSelectionInterface.php) returns the SSA result for the specified `inventoryRequest`. 
+
+## Develop a custom algorithm
+
+As you develop your custom Source Selection Algorithm, keep these design considerations in mind:
+
+* Implement `SourceSelectionInterface`
+* If your module provides an SSA on quotes, introduce your own `InventoryRequestFactory`
+* Register your SSA within a `di.xml` file
+
+### Implement  `SourceSelectionInterface` 
+
+Your SSA must implement `SourceSelectionInterface`, which is shown below:
+
+```php?start_inline=1
+/**
+ * Returns source selection algorithm result for given Inventory Request 
+ * Current interface should be implemented in order to add own Source Selection Method
+ * 
+ * @api
+ */
+interface SourceSelectionInterface
+{
+    /**
+     * @param InventoryRequestInterface $inventoryRequest
+     * @return SourceSelectionResultInterface
+     */
+    public function execute(
+       InventoryRequestInterface $inventoryRequest
+    ): SourceSelectionResultInterface;
+}
+````
+
+For example, the following example defines the `MinimalDeliveryCostAlgorithm` class:
+
+```php?start_inline=1
+namespace Some\Vendor\Namespace\SourceSelection;
+
+/**
+ * Minimal Delivery Cost for Merchant algorithm
+ *
+ * @api
+ */
+class MinimalDeliveryCostAlgorithm implements SourceSelectionInterface
+{
+    public function execute(
+       InventoryRequestInterface $inventoryRequest
+    ): SourceSelectionResultInterface;
+    {
+        // TODO: Implement execute() method.
+    }
+}
+```
+
+### Create a `InventoryRequest` factory for quotes  (optional)
+
+Magento provides the [`InventoryRequestFromOrderFactory`](https://github.com/magento-engcom/msi/blob/2.3.0-release/app/code/Magento/InventoryShipping/Model/InventoryRequestFromOrderFactory.php), which determines which sources to use to fulfill the order at the time a shipment is created. 
```

Comment:
which determines which? Is this like where oh where is my grammar book?

Reply:
Got it. Changed both instances of "which" to "that". 

J/K. I replaced the second "which" to "the"